### DISCUSSION TITLE
Setting color when using datum

### DISCUSSION THREAD

**Discussion by jankuca** [vega/vega-lite#8499] [Q&A]
Hello,

I am trying to figure out a combination of `datum` and some custom color configuration but I cannot seem to get to one.

Is it possible to override the automatically selected colors when using `encoding.color.datum`?

Here is an example vega spec of one of my tries: [setting `mark.color`](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoMcGmSxoATBoAMSpA0xsINAF5w0oTAE8c59ADM6IAL5KAJkkwpUoYkkEMcBBoANqggp5oAIxKgnKaSgDumq7hkagxIHHyqGJJaABsqVnpGrHxqADM+agAnM4Auq4lVnAATqGgyG0A1hYg1rZoIExIHUqGbIKYNDhomG2BSlBTbB3obXBuLkpwsituNLJk-QAe-Y5wgtvoEZiKAzZ2ILJsCEf+OyBWFzRXN1l4hMnsMAI4MJCyGZeGakL4rOLrUAeTAMBDDCJMK4uZyNZxAA)

Another approach I took is [specifying the `encoding.color.scale.scheme`](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoMcGmSxoATBoAMSpA0xsINAF5w0oTAE8c59ADM6IAL5KAJkkwpUoYkkEMcBBoANqggp5oAIxKgnKaSgDumq7hkagxIHHyqGJJaABsqVnpGrHxqADM+agAnM4Auq4lVnAATqGgyG0A1hYg1rZoIExIHUqGbIKYNDhomG2BzXCyUGxuNLJk-QAe-Y5wgm7DEZiKAzZ2ILJsCJv+LkpW+zSHx+jZ54NXAI4MSLIZl4ZqRHiA1nEOj4QB5MAwECckExDudoP47KAuIgrm04G5gs5CY1nEA) but that gives a warning about the unexpected use of `datum` in such case.

Is there any other way?

Any help much appreciated, thanks!

**Comment by jwoLondon** [comment]
It's not quite clear to me what you are wanting to achieve here. In your first example you correctly set the colour of all bar marks to `"red"`, but then you also add a further colour encoding with `datum:"label"` which is not necessary and is preventing the bars from being assigned a red colour. Simply removing your colour encoding line (` "color": {"datum": "label"}`) will solve that particular problem.

In your second example, are you wanting the value of `w` to influence the colour of the mark that represents it? If so, you'd replace `datum:"label"` with  `field:"w"`. This is the standard way in which you indicate which data you want to influence the encoding.

It is possible to be more precise in colour allocation than just using a colour scheme, but whether you go down that route I think depends on what exactly you want to achieve.
